oak_sign/src/main.rs

```diff
@@ -152,11 +161,13 @@ fn main() -> anyhow::Result<()> {
                     .expect("Couldn't parse PKCS8 encoded private key");
             let public_key_bytes = key_pair.public_key();
             let signature_bytes = key_pair.sign(&input_file_bytes);
```

Comment:
Is it wise to generate a hash, store that in the signature file, but then calculate the actual signature on the original input bytes instead of the computed hash (which will presumably still sign the hash, though it may or may not be the same hash)? It seems it may make more sense to just sign the computed hash (even if it ends up being re-hashed as part of the signature)?

@benlaurie do you have an opinion on this?

Reply:
[Ed25519](https://en.wikipedia.org/wiki/EdDSA#Ed25519) already uses SHA-512 for signing.
I think if we will start signing SHA-256 hashes with Ed25519 it may raise a lot of questions from the Oak application developers in the future.

Reply:
> Is it wise to generate a hash, store that in the signature file, but then calculate the actual signature on the original input bytes instead of the computed hash

Hash that is stored in the file is used just as a reference, so when we want to check a Wasm file's signature, we compute its hash, and then find a corresponding file, that contains the same hash.

Reply:
Let's follow up separately, so we don't block this PR. Could you create a separate issue to track, so we don't forget?

Reply:
Created an issue https://github.com/project-oak/oak/issues/1437